### PR TITLE
Consider team group conversations with one participant 1-to-1

### PR DESCRIPTION
**Pull request by typfel**
We want to consider team group conversations with only 2 participants to be one-to-one. This means supporting the same features (delivery recipes, video calling) and design.

see: https://github.com/wireapp/architecture/blob/master/teams/clients.md#one-to-one-conversation-inside-teams

### CODE REVIEW

File: Source/Model/Conversation/ZMConversation.m [hunk 1]
```diff
@@ -473,6 +481,24 @@ - (void)setUserDefinedName:(NSString *)aName {
...
+        conversationType = ZMConversationTypeOneOnOne;
+    }
+    
+    [self didAccessValueForKey:ZMConversationConversationTypeKey];
+    return conversationType;
+}
+
+- (ZMConversationType)internalConversationType
+{
+    return (ZMConversationType)[[self primitiveConversationType] shortValue];
```
Comment (marcoconti83) [inline Source/Model/Conversation/ZMConversation.m]: Is `primitiveConversationType` a new CoreData property? I don't see it being added to the CoreData model in this PR.
Here you call `[self willAccessValueForKey:ZMConversationConversationTypeKey]` but the actual CoreData value for `ZMConversationConversationTypeKey` is never accessed. Unless this points to `primitiveConversationType`.

Comment (typfel) [inline Source/Model/Conversation/ZMConversation.m]: `primitiveConversationType` is not a new CoreDara property but is a getter generated by CoreData for accessing the internal storage of `conversationType` (see https://developer.apple.com/documentation/coredata/nsmanagedobject/1506728-primitivevalueforkey)

on ZMConversation.m:145 we declare several other primitive properties which are used in a similar manner.

Comment (marcoconti83) [inline Source/Model/Conversation/ZMConversation.m]: Thx for explaining, maybe the `willAccessValue/didAccessValue` should be inside this method then.

**Comment by marcoconti83** [comment]
Looks good now!

**Pull request merged**